src/polymer/polymer-element.ts

```diff
@@ -239,53 +239,152 @@ function propertyToAttributeName(propertyName: string): string|null {
 
 function resolveElement(
     scannedElement: ScannedPolymerElement, document: Document): PolymerElement {
-  // TODO: Copy over all properties better. Maybe exclude known properties not
-  //   copied?
-  const clone: PolymerElement =
-      Object.assign(new PolymerElement(), scannedElement);
+  const element = new PolymerElement();
+
+  //
+  // Superclass
+  //
+  if (scannedElement.superClass && scannedElement.superClass != 'HTMLElement') {
+    const superElements =
+        document.getById('element', scannedElement.superClass, {
+          externalPackages: true,
+          imported: true,
+        });
+    if (superElements.size === 1) {
+      const superElement = superElements.values().next().value;
+      if (!superElement.kinds.has('polymer-element')) {
+        element.warnings.push({
+          message:
+              `A Polymer element can\'t extend from a non-Polymer element: ${scannedElement
+                  .superClass}`,
+          severity: Severity.ERROR,
+          code: 'unknown-superclass',
+          // TODO(justinfagnani): we need to know where the superclass came
+          // (annotation vs extends clause) from and track its range.
+          sourceRange: null as any as SourceRange,
```

Comment:
This will cause a lot of unknown problems down the line. Mind modifying the superClass object to track both name and the source range of the reference?